### PR TITLE
Abstracts, General and canonical types for TK

### PR DESCRIPTION
**Pull request by RathishKJ**

### CODE REVIEW

**Review by darshanjain12 — COMMENTED** [review]
Hello Ratish i guess you have by mistakely wrote "ahich" in TK  description

**Comment by RathishKJ** [comment]
> Hello Ratish i guess you have by mistakely wrote "ahich" in TK description

@darshanjain12 , Thanks for noticing the error... It has been updated...

File: ontology/yaml/resources/HVAC/entity_types/TK.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  - PHWTC
+  
+TK_TLM:  
+  description: "Tank with water level monitoring"
+  is_canonical: true
+  implements:
+  - TK
+  - TLM 
+
+TK_SPR_PLPM:
```
Comment (tasodorff) [inline ontology/yaml/resources/HVAC/entity_types/TK.yaml:36]: Remove sprinkler reference, its unnecessary.

File: ontology/yaml/resources/HVAC/entity_types/TK.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+TK_HDR_PLPM:  
+  description: "Tank with pipeline pressure monitoring for hydrant tank pipeline"
+  is_canonical: true
+  implements:
+  - TK
+  - PLPM  
+
+TK_FHT_TLM:  
+  description: "Tank with level monitoring for fire hydrant tanks"
```
Comment (tasodorff) [inline ontology/yaml/resources/HVAC/entity_types/TK.yaml:51]: same as above, not important if it serves fire hydrants. however, it should be in the SAFETY namespace.

**Review by tasodorff — CHANGES_REQUESTED** [review]
remove use-case specifics, but do need to move anything fire suppression related to the SAFETY namespace. I suggest defining tank at the global level and referencing it for a local defininition of TK in both the HVAC and SAFETY namespaces.

File: ontology/yaml/resources/HVAC/entity_types/TK.yaml (hunk 1 above)
Comment (RathishKJ) [reply ontology/yaml/resources/HVAC/entity_types/TK.yaml]: Changed from TK_SPR_PLPM  to TK_PLPM

File: ontology/yaml/resources/HVAC/entity_types/TK.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  - TLM 
+
+TK_SPR_PLPM:  
+  description: "Tank with pipeline pressure monitoring for sprinkler pipeline"
+  is_canonical: true
+  implements:
+  - TK
+  - PLPM
+
+TK_HDR_PLPM:
```
[the remark this answers is not in the document]
Comment (RathishKJ) [reply ontology/yaml/resources/HVAC/entity_types/TK.yaml]: Removed TK_HDR_PLPM

File: ontology/yaml/resources/HVAC/entity_types/TK.yaml (hunk 2 above)
Comment (RathishKJ) [reply ontology/yaml/resources/HVAC/entity_types/TK.yaml]: Moved to FACP pull request #634

**Comment by RathishKJ** [comment]
> remove use-case specifics, but do need to move anything fire suppression related to the SAFETY namespace. I suggest defining tank at the global level and referencing it for a local defininition of TK in both the HVAC and SAFETY namespaces.

TANK has been defined in the global name space and the same is being pulled for HVAC and SAFETY. Fire hydrant tank has been moved to  SAFETY.

File: ontology/yaml/resources/HVAC/entity_types/TK.yaml (hunk 1 above)
Comment (RathishKJ) [reply ontology/yaml/resources/HVAC/entity_types/TK.yaml]: deleted the canonical...

File: ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml [hunk 4]
```diff
@@ -4979,3 +4979,27 @@ EFHLC:
   implements:
   - OPERATIONAL
 
+  
+PHWTC:
+  description: "portable water temperature monitoring and control."
```
Comment (tasodorff) [inline ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml:4984]: typo, fix

File: ontology/yaml/resources/fields/telemetry_fields.yaml [hunk 5]
```diff
@@ -1323,10 +1323,16 @@ literals:
   - ON
   - OFF
   
+
+#Addition of new fields for Tanks##
+- level_status
```
Comment (tasodorff) [inline ontology/yaml/resources/fields/telemetry_fields.yaml:1328]: what enum states? list them.

File: ontology/yaml/resources/HVAC/entity_types/GENERALTYPES.yaml [hunk 6]
```diff
@@ -365,3 +365,7 @@ GTWS:
   - differential_pressure_specification
   - flowrate_requirement
 
+
```
Comment (tasodorff) [inline ontology/yaml/resources/HVAC/entity_types/GENERALTYPES.yaml:368]: revert if you are just adding whitespace.

File: ontology/yaml/resources/entity_types/global.yaml [hunk 7]
```diff
@@ -53,4 +53,11 @@ INCOMPLETE:
   id: "4665780478685478912"
   description: "A known incomplete type."
   is_abstract: true
+  
+##Global namespace for tanks##
+TANK:
```
Comment (tasodorff) [inline ontology/yaml/resources/entity_types/global.yaml:58]: should conform to [BDNS](https://github.com/theodi/BDNS/blob/master/BDNS_Abbreviations_Register.csv) unless there is a good reason not to. think `TK` is appropriate for a generic tank.

**Review by tasodorff — CHANGES_REQUESTED** [review]

File: ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml (hunk 4 above)
Comment (RathishKJ) [reply ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml:4984]: Changed to "potable"

File: ontology/yaml/resources/fields/telemetry_fields.yaml (hunk 5 above)
Comment (RathishKJ) [reply ontology/yaml/resources/fields/telemetry_fields.yaml:1328]: Added Low, Medium and High as the states in the telemetry

File: ontology/yaml/resources/entity_types/global.yaml (hunk 7 above)
Comment (RathishKJ) [reply ontology/yaml/resources/entity_types/global.yaml:58]: Replaced the TANK to TK in the global namespace.

File: ontology/yaml/resources/HVAC/entity_types/TK.yaml (hunk 2 above)
Comment (RathishKJ) [reply ontology/yaml/resources/HVAC/entity_types/TK.yaml]: Created the canonical under safety TK_LM for fire hydrant tank level monitoring.

**Review by tasodorff — APPROVED** [review]

**Pull request merged**